File: src/com/sterndu/data/transfer/basic/ServerSocket.java
package com.sterndu.data.transfer.basic;

import java.io.IOException;
import java.net.*;
import com.sterndu.data.transfer.DatatransferServerSocket;

public class ServerSocket extends DatatransferServerSocket {

	public ServerSocket() throws IOException {}

	public ServerSocket(int port) throws IOException {
		super(port);
	}

	public ServerSocket(int port, int backlog) throws IOException {
		super(port, backlog);
	}

	public ServerSocket(int port, int backlog, InetAddress bindAddr) throws IOException {
		super(port, backlog, bindAddr);
	}

	@Override
	public Socket accept() throws IOException {
		final Socket s = new Socket();
		super.implAccept(s);
		s.host = true;
		s.init(true);
		return s;
	}

}

File: src/com/sterndu/data/transfer/basic/ServerSocket.kt
@file:JvmName("ServerSocket")
package com.sterndu.data.transfer.basic

import java.io.IOException
import java.net.ServerSocket

open class ServerSocket(val serverSocket: ServerSocket = ServerSocket()) {

	@Throws(IOException::class)
	fun accept(): Socket {
		val s = try {
			serverSocket.accept()
		} catch (e: IOException) {
            throw e
		}
		return Socket(s, false).apply {
			initWithHost(true)
		}
	}
}
